peers/prefer_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package peers
+
+import (
+	"fmt"
+	"hash/fnv"
+	"testing"
+	"time"
+
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/raw"
+	"github.com/uber/tchannel-go/testutils"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uber-go/atomic"
+	"golang.org/x/net/context"
+)
+
+func TestHRWScorerGetScore(t *testing.T) {
+	client := testutils.NewClient(t, nil)
+	peer := client.Peers().GetOrAdd("1.1.1.1")
+
+	c1 := NewHRWScorer(1)
+	c2 := NewHRWScorer(2)
+
+	assert.NotEqual(t, c1.GetScore(peer), c2.GetScore(peer))
+}
+
+func TestHRWScorerDistribution(t *testing.T) {
+	const (
+		numClients = 1000
+		numServers = 20
+	)
+
+	ch := testutils.NewClient(t, nil)
+	servers := make([]*tchannel.Peer, numServers)
+	for i := range servers {
+		servers[i] = ch.Peers().GetOrAdd(fmt.Sprintf("1.1.1.%v", i))
+	}
+
+	serverSelected := make([]int, numServers)
+	for i := 0; i < numClients; i++ {
+		client := NewHRWScorer(uint32(i))
+
+		highestScore := uint64(0)
+		highestServer := -1
+		for s, server := range servers {
+			if score := client.GetScore(server); score > highestScore {
+				highestScore = score
+				highestServer = s
+			}
+		}
+		serverSelected[highestServer]++
+	}
+
+	// Validate that we have a reasonable distribution. We expect 50 +- 20.
```

Comment:
Updated the test to use the constants and added comment about tolerance, let me know if the comment answers your question.